contracts/contracts/ValidatorAuction.sol

```diff
@@ -135,15 +146,44 @@ contract ValidatorAuction is Ownable {
 
     function withdraw() public {
         require(auctionState == AuctionState.Ended || auctionState == AuctionState.Failed, "You cannot withdraw before the auction is ended or it failed.");
-        require(bids[msg.sender] > closingPrice, "The sender has nothing to withdraw.");
+        require(bids[msg.sender] >= lowestBidPrice, "The sender has nothing to withdraw.");
```

Comment:
This is our invariant, so if this fails, we screwed up, so I would say that if it is there this should not be a require but an assert, and the text is wrong. To check if someone can withdraw, we have later the specific checks. 
I would remove this all together.